src/fmod.c

```diff
@@ -1,10 +1,14 @@
 #include <math.h>
 
 double fmod(double x, double div) {
-	if(div == 0) {
-		return 0;
+	//if div == nan return x directly
+	if (div != div) {
+		return x;
 	}
-	while(x > div) {
+	if (div == 0) {
+		return 0.0 / 0.0;
```

Comment:
Is there some other way to return nan here? Compiling this gives a division by zero error.